### DISCUSSION TITLE
Consistency when using output trigger bindings with Durable Entities

### DISCUSSION THREAD

**Discussion by mpaul31** [Azure/azure-functions-durable-extension#2318] [Q&A]
@sebastianburckhardt or others.

When using output triggers (service bus, event grid, etc.) within a durable entity, what does the consistency and durability look like? For instance, I have a Bid entity and after I update the entity state I would like to publish an event using the event hubs output trigger. I'm just trying to understand the order of operations and what happens when the publishing of the event fails or committing the durable state to the storage provider. Would it be possible when one of these situations arises duplicate events could be published? Any information or guidance would be appreciated!

**Comment by mpaul31** [comment]
@sebastianburckhardt bump :)

**Reply by davidmrdavid** [reply]
Just a heads up - @sebastianburckhardt is out for a few days so it might take him a tad to respond to this thread.
My personal understanding (which may be off) is that the updating of entity state is quite independent of the output binding succeeding, and that this means that updating external stores need to be handled with special care. We've been thinking about improving support for dealing with these failure scenarios, but I'll leave it to @sebastianburckhardt to elaborate on the details.
